tests/shared/src/test/scala/scala/meta/tests/parsers/TypeSuite.scala

```diff
@@ -167,32 +188,43 @@ class TypeSuite extends ParseSuite {
   }
 
   test("a.T forSome { val a: A }") {
-    val Existential(
-      Select(TermName("a"), TypeName("T")),
-      Decl.Val(Nil, Pat.Var(TermName("a")) :: Nil, TypeName("A")) :: Nil
-    ) =
-      tpe("a.T forSome { val a: A }")
+    assertTpe("a.T forSome { val a: A }")(
+      Existential(
+        Select(TermName("a"), TypeName("T")),
+        Decl.Val(Nil, Pat.Var(TermName("a")) :: Nil, TypeName("A")) :: Nil
+      )
+    )
   }
 
   test("A | B is not a special type") {
-    val comp @ ApplyInfix(TypeName("A"), TypeName("|"), TypeName("B")) = tpe("A | B")
+    assertTpe("A | B")(ApplyInfix(TypeName("A"), TypeName("|"), TypeName("B")))
   }
 
   test("42.type") {
     intercept[ParseException] {
       tpe("42")(dialects.Scala211)
     }
 
-    val Lit(42) = tpe("42")(dialects.Scala3)
-    val Lit(-42) = tpe("-42")(dialects.Scala3)
-    val Lit(42L) = tpe("42L")(dialects.Scala3)
-    val Lit(42f) = tpe("42f")(dialects.Scala3)
-    val Lit(-42f) = tpe("-42f")(dialects.Scala3)
-    val Lit(42d) = tpe("42d")(dialects.Scala3)
-    val Lit(-42d) = tpe("-42d")(dialects.Scala3)
-    val Lit("42") = tpe("\"42\"")(dialects.Scala3)
-    val Lit(false) = tpe("false")(dialects.Scala3)
-    val Lit(true) = tpe("true")(dialects.Scala3)
+    implicit val dialect = dialects.Scala3
+
+    def matchSubStructureTyp3(typ: String, func: PartialFunction[Tree, Unit]) = {
+      matchSubStructure[Type](typ, func)(
+        parseType,
+        dialects.Scala3,
+        implicitly[munit.Location]
```

Comment:
should you pass Location as implicit to `...Typ3`? or just declare scala3 as implicit dialect, and use `matchSubStructure[Type](...)` directly?

Reply:
Ach, yes. Added it as implicit.